### DISCUSSION TITLE
[error]NO PLANNING LIBRARY LOADED

### DISCUSSION THREAD

**Discussion by starROS** [moveit/moveit2#2297] [Q&A] [answered]
After performing a recent update (sudo apt update), I encountered multiple problems while using MoveIt. These problems were characterized by error messages such as "RTPS_MSG_IN Problem reserving CacheChange" and "NO PLANNING LIBRARY LOADED." As a response, we decided to uninstall ROS and then reinstall it.

To assess whether the issues were resolved, we employed a demo launch file from the MoveIt tutorials package. However, I still observed the "NO PLANNING LIBRARY LOADED" error within the rviz tool. Additionally, the robot model displayed erratic behavior, oscillating between two states.

It's worth noting that this inconsistent behavior sometimes disappears. On occasion, the robot model loads successfully and the OMPL (Open Motion Planning Library) functions without any errors.

For reference, our system details are as follows: Ubuntu 22.04 with ROS2 Humble

**Comment by bmaxdk** [comment, marked as answer]
`RTPS_MSG` may be related issue could happen. If you use different DDS implementations, it may cause RTPS errors. Have you tried to reinstalled ROS2 humble moveit2?

You may also want to switch between different DDS implementations.
```bash
$ export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
```

**Reply by starROS** [reply]
Thanks
